### PR TITLE
Added missing class summaries for YWidget children.

### PR DESCRIPTION
**Pull request by mvidner**
For the Doxygen documentation (which really should get a public online rendering)

### CODE REVIEW

File: src/YBarGraph.h [hunk 1]
```diff
@@ -32,7 +32,11 @@
 class YBarGraphPrivate;
 class YBarGraphSegment;
 
-
+/**
+ * A graph showing partitioning of a whole.
+ * The whole is divided into YBarGraph segments,
+ * each of which has a relative size, a color, and a label.
```
Comment (jreidinger) [inline src/YBarGraph.h]: I think link to class below would be nice and there is also inconsistency as below you mention two colors ( label and background) and here only generic color.

File: src/YCheckBoxFrame.h [hunk 2]
```diff
@@ -32,6 +32,9 @@
 class YCheckBoxFramePrivate;
 
 
+/**
+ * A frame with a check-box, may auto-disable frame contents based on the check.
```
Comment (jreidinger) [inline src/YCheckBoxFrame.h]: may? is there any rule when it is auto-disabled?

File: src/YLayoutBox.h [hunk 3]
```diff
@@ -31,7 +31,9 @@
 
 class YLayoutBoxPrivate;
 
-
+/**
+ * A vertical or horizontal stacking of widgets, implementing HBox and VBox.
```
Comment (jreidinger) [inline src/YLayoutBox.h]: a bit misleading to talk about its usage in different project as HBox and VBox. I think note about H/VBox is not needed.

File: src/YReplacePoint.h [hunk 4]
```diff
@@ -27,6 +27,9 @@
 
 #include "YSingleChildContainerWidget.h"
 
+/**
+ * The only widget allowed to act as a parent in a ReplaceWidget operation.
```
Comment (jreidinger) [inline src/YReplacePoint.h]: I think better start with its usage. Something like "Placeholder widget used for dynamic changing its content by other one"

File: src/YRichText.h [hunk 5]
```diff
@@ -33,6 +33,9 @@
 class YRichTextPrivate;
 
 
+/**
+ * Text formatted with simple HTML-like tags, with "links" generating events.
```
Comment (jreidinger) [inline src/YRichText.h]: any pointer to supported tags would be really nice.

**Comment by jreidinger** [comment]
just few ideas how to improve it. In general better then before so LGTM

File: src/YLayoutBox.h (hunk 3 above)
Comment (mvidner) [inline src/YLayoutBox.h]: Well, yes, but other classes mention the YCP names too. We may clarify this in a future cleanup.

File: src/YRichText.h (hunk 5 above)
Comment (mvidner) [inline src/YRichText.h]: Yes, but it is out of scope for my current plan "Add missing one line descriptions to all classes".

File: src/YCheckBoxFrame.h (hunk 2 above)
Comment (mvidner) [inline src/YCheckBoxFrame.h]: linked to the related methods

**Comment by mvidner** [comment]
Thank you for the review!

**Pull request merged**